cmd/regres/main.go

```diff
@@ -0,0 +1,239 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Regress is a tool to display build and runtime statistics over a range of
+// changelists.
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"runtime"
+	"text/tabwriter"
+	"time"
+
+	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/git"
+	"github.com/google/gapid/core/log"
+	"github.com/google/gapid/core/os/shell"
+)
+
+var (
+	root     = flag.String("root", "", "Path to the root GAPID source directory")
+	verbose  = flag.Bool("verbose", false, "Verbose logging")
+	optimize = flag.Bool("optimize", false, "Build using '-c opt'")
+	output   = flag.String("out", "", "The results output file. Empty writes to stdout")
+	count    = flag.Int("count", 2, "The number of changelists to profile since HEAD")
+)
+
+func main() {
+	app.ShortHelp = "Regress is a tool to perform performance measurments over a range of CLs."
+	app.Run(run)
+}
+
+type stats struct {
+	SHA                  string
+	BuildTime            float64  // in seconds
+	IncrementalBuildTime float64  // in seconds
+	FileSizes            struct { // in bytes
+		LibGAPII                   int64
+		LibVkLayerVirtualSwapchain int64
+		GAPIDAarch64APK            int64
```

Comment:
In addition to tracking the total APK size, I think it would be really useful to track the size of the individual libs inside the APKs.

Reply:
Good idea. Will consider adding in a followup CL.